app/controllers/main/routes.py

```diff
@@ -47,50 +49,134 @@ def events(selectedTerm=None):
+
+              noteForDict = list(notes)[-1].banNote.noteContent if list(notes) else ""
+              eligibilityTable.append({"program" : program,
+                                   "completedTraining" : (username in trainedParticipants(program)),
+                                   "isNotBanned" : isEligibleForProgram(program, username),
+                                   "banNote": noteForDict})
+         return render_template ("/main/volunteerProfile.html",
+            # programs = programs,
+            # interests = interests,
+            programsInterested = programsInterested,
+            upcomingEvents = upcomingEvents,
+            eligibilityTable = eligibilityTable,
+            volunteer = User.get(User.username == username),
+            backgroundTypes = backgroundTypes,
+            completedBackgroundCheck = completedBackgroundCheck
+            )
+    abort(403)
+
+
+@main_bp.route('/<username>/ban/<program_id>', methods=['POST'])
+def ban(program_id, username):
+    """
+    This function updates the ban status of a username either when they are banned from a program.
+    program_id: the primary id of the program the student is being banned from
+    username: unique value of a user to correctly identify them
+    """
+    postData = request.form
+    banNote = postData["note"] # This contains the note left about the change
+    banEndDate = postData["endDate"] # Contains the date the ban will no longer be effective
     try:
-        profileUser = User.get(User.username == username)
-        upcomingEvents = getUpcomingEventsForUser(username)
-        programs = Program.select()
-        interests = Interest.select().where(Interest.user == profileUser)
-        interests_ids = [interest.program.id for interest in interests]
-        rsvpedEventsList = EventRsvp.select().where(EventRsvp.user == profileUser)
-        rsvpedEvents = [event.event.id for event in rsvpedEventsList]
-
-        allUserEntries = list(BackgroundCheck.select().where(BackgroundCheck.user == profileUser))
-        completedBackgroundCheck = {entry.type.id: entry.passBackgroundCheck for entry in allUserEntries}
-        backgroundTypes = list(BackgroundCheckType.select())
-
-
-        return render_template('/volunteer/volunteerProfile.html',
-                               title="Volunteer Interest",
-                               user = profileUser,
-                               programs = programs,
-                               interests = interests,
-                               interests_ids = interests_ids,
-                               upcomingEvents = upcomingEvents,
-                               rsvpedEvents = rsvpedEvents,
-                               backgroundTypes = backgroundTypes,
-                               completedBackgroundCheck = completedBackgroundCheck
-                               )
+        flash("Successfully banned user", "success")
```

Comment:
Don't flash before you actually try it. If you do this and there is an exception in `banUser`, the successful flash has already happened.